### PR TITLE
Jump point convenience fields

### PR DESCRIPTION
**Pull request by nloewen**
Add some convenience fields to `JumpPoint`.

`JumpPoint.returnJump`
Adding a pointer to the return jump allows us to avoid looping over jump points in a number of places. The looping still needs to happen, but we only need to do in once after our memory allocations are finished.

`JumpPoint.from`
The reference to the containing system is not yet used, but will be in an upcoming PR.The idea is that you'll be able to perform an operation such as an adjacency search, and get access to the relevant `JumpPoint` and `StarSystem` in one struct. This will be a bigger deal once hypergates are added, which will introduce adjacent systems that aren't directly reachable through `system->jumps[i].target`.

### CODE REVIEW

File: src/space.h [hunk 1]
```diff
@@ -184,7 +187,7 @@ typedef struct JumpPoint_ {
    double sina; /**< Sinus of the angle. */
    int sx; /**< X sprite to use. */
    int sy; /**< Y sprite to use. */
-} JumpPoint;
+};
```
Comment (onpon4) [inline src/space.h:190]: Why did you remove the usual `typedef struct Foo_ { ... } Foo;` idiom? It's the usual way to do these.

🕵️

Comment (nloewen) [reply src/space.h:190]: It needs to be forward declared in order to use it in the struct definition

**Comment by bobbens** [comment]
Looks good, thanks!
